src/NBomber/Plugins/PsPingPlugin.fs

```diff
@@ -10,27 +10,37 @@ open FSharp.Control.Tasks.NonAffine
 open FsToolkit.ErrorHandling
 open Microsoft.Extensions.Configuration
 
+open NBomber
 open NBomber.Contracts
+open NBomber.Domain.Stats.Statistics
 open NBomber.Extensions.InternalExtensions
 
 [<CLIMutable>]
 type PsPingPluginConfig = {
     Hosts: Uri[]
     /// The default is 1000 ms.
     Timeout: int
+    /// Number of ping executions. The default is 4.
+    Executions: int
 } with
-    static member CreateDefault([<ParamArray>]hosts: string[]) = {
+    static member CreateDefault([<ParamArray>]hosts: string[]) =
+        {
             Hosts = hosts |> Array.map Uri
             Timeout = 1_000
-    }
+            Executions = 4
```

Comment:
let's simplify it
let's don't have it configurable but rather use it as a constant value
let's have 3 only invocations not 4

Reply:
I need to have this configurable, and to be able to increase the value, as sometimes there is a very high variability in the psping values (see my previous reply) ...

Reply:
Maybe we should run it at the start of test plus and end of the test to capture a bigger diapason?

Reply:
on the other hand, I think all this brings an accidental complexity.
This is not the primary purpose of NBomber to correctly measure PING

Reply:
Hi @deyanp 
I have created a new project NBomber.Contracts
https://github.com/PragmaticFlow/NBomber/tree/v3/src/NBomber.Contracts

Reply:
Hi @AntyaDev , not sure what that brings, is it in relation to the discussion about Plugins depending on some InternalExtensions?

It turned out I just need these, so I copy-pasted them in my local project and managed to decouply the PsPingPlugin from the NBomber project:

```
namespace Framework.NBomber.Extensions

module InternalExtensions =

    let inline isNotNull (value) =
        not(isNull value)

    module Option =

        let ofRecord (value: 'T) =
            let boxed = box(value)
            if isNotNull(boxed) then Some value
            else None
```
and

```
module internal Framework.NBomber.Domain.Stats.Statistics

open System

module Constants =
    let StatsRounding = 2

module Converter =
    let inline round (digits: int) (value: float) = Math.Round(value, digits)
```

I have adjusted a bit the code so that there are also a few "warmup" pings at the beginning, as otherwise I was getting some crazy ping durations ... Still getting such for Azure though, not sure why, even though I am pinging from 1 AKS cluster another AKS cluster both the same vnet and region (the legendary 2ms are not achievable ;). Same setup in GCP is much more stable for some reason ...

Reply:
The point with NBomber.Contracts is that to develop any plugin; you need to add reference on NBomber.Contracts instead of the complete NBomber projects. In this way, you don't create cycle references. In other words, you fully decoupled.

Reply:
Regarding 2ms and customization around warmup - from my perspective is a good sign that something went wrong.
But ok, I explained it already several times :)